### PR TITLE
add kinesis/kintlc

### PR DESCRIPTION
**Pull request by aleb**
Added support for Teensy LC to the kinx/kint board for Kinesis keyboards.

## Description

I added a new keyboard by copy/pasting from kinesis/kint36, kinesis/kint41 and handwired/onekey/teensy_lc. 

I tested with a Kinesis KB500 with the dev/master Stapelberg board with a Teensy LC, works fine.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://github.com/kinx-project/kint/issues/1

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kinesis/kint36/rules.mk [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# MCU name
+# MCU name from https://docs.qmk.fm/#/compatible_microcontrollers
```
Comment (fauxpark) [inline keyboards/kinesis/kint36/rules.mk:1]:
```suggestion
# MCU name
```

File: keyboards/kinesis/kintlc/rules.mk [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# MCU name from https://docs.qmk.fm/#/compatible_microcontrollers
```
Comment (fauxpark) [inline keyboards/kinesis/kintlc/rules.mk:1]:
```suggestion
# MCU name
```

File: keyboards/kinesis/kintlc/rules.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+BOOTLOADER = halfkay
+
+# Debounce eagerly (report change immediately), keep per-key timers. We can use
+# this because the Cherry MX keyswitches on the Kinesis only produce noise while
+# pressed.
+DEBOUNCE_TYPE = sym_eager_pk
+
+# Specify firmware format that is copied to root qmk_firmware folder
+FIRMWARE_FORMAT = hex
```
Comment (fauxpark) [inline keyboards/kinesis/kintlc/rules.mk:14]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by aleb** [comment]
I've been using it for the past three weeks with no issues.

File: keyboards/kinesis/kint36/rules.mk [hunk 4]
```diff
@@ -5,5 +5,5 @@ MCU = MK66FX1M0
 # this because the kinT does not have to deal with noise.
 DEBOUNCE_TYPE = sym_eager_pk
 
-#Specify firmware format that is copied to root qmk_firmware folder
+# Specify firmware format that is copied to root qmk_firmware folder
 FIRMWARE_FORMAT = hex
```
Comment (fauxpark) [inline keyboards/kinesis/kint36/rules.mk:9]:
```suggestion
```
This should not be needed as of #17014

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kinesis/kint41/rules.mk [hunk 5]
```diff
@@ -3,13 +3,10 @@ MCU_SERIES = MIMXRT1062
...
 MCU_STARTUP = MIMXRT1062
 BOARD = IC_TEENSY_4_1
-MCU  = cortex-m4
+MCU = cortex-m4
 ARMV = 7
 
 # Debounce eagerly (report change immediately), keep per-key timers. We can use
 # this because the Cherry MX keyswitches on the Kinesis only produce noise while
 # pressed.
 DEBOUNCE_TYPE = sym_eager_pk
```
Comment (fauxpark) [inline keyboards/kinesis/kint41/rules.mk:12]:
```suggestion
DEBOUNCE_TYPE = sym_eager_pk

FIRMWARE_FORMAT = hex
```
Teensy 4.1 does not have an entry in mcu_selection.mk yet so it is not covered here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**